src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,219 @@
+        }
+        return users;
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET user_name = ?, login = ?, password = ? "
+                + "WHERE user_id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update user with id = "
+                    + user.getId(), ex);
+        }
+
+        removeUserRoles(user.getId());
+        insertUserRoles(user);
+        return user;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE users SET deleted = TRUE WHERE user_id = ?;";
+        int itemsDeleted = 0;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            itemsDeleted = statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete product with id = " + id, ex);
+        }
+        return (itemsDeleted == 1 && removeUserRoles(id));
+    }
+
+    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
+        Long userId = resultSet.getLong("user_id");
+        String userName = resultSet.getString("user_name");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+        return new User(userId, userName, login, password);
+    }
+
+    private Set<Role> getUserRolesFromResultSet(Long userId) {
+        String query = "SELECT ur.role_id AS id, role_name AS name "
+                + "FROM users_roles ur "
+                + "INNER JOIN roles r ON ur.role_id = r.role_id "
+                + "WHERE user_id = ?";
+        Set<Role> roles = new HashSet<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Role role = Role.of(resultSet.getString("name"));
+                role.setId(resultSet.getLong("id"));
+                roles.add(role);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get roles of user with id = "
+                    + userId, ex);
+        }
+        return roles;
+    }
+
+    private boolean insertUserRoles(User user) {
```

Comment:
I understand that this method is bad implemented according to point 8 in the checklist. But I can't figure out how to refactor it. Please give me a hint.

Reply:
Maybe try to split it on 2 methods?

Reply:
I did it in one statement with SELECT function in VALUES() 
INSERT INTO users_roles (user_id, role_id) SELECT ?, "
                + " role_id FROM roles WHERE role_name = ?;"

Reply:
you can user select subquery

Reply:
Thanks for your advices! Variant with subquery is much shorter and I had no practice with subqueries before so I've chosen it)